ui/src/app/job-list/table/table.component.spec.ts

```diff
@@ -97,59 +103,77 @@ describe('JobsTableComponent', () => {
         SharedModule
       ],
       providers: [
-        {provide: JobManagerService, userValue: newDefaultMockJobManagerService()}
+        {provide: JobManagerService, userValue: new FakeJobManagerService([])}
       ],
     }).compileComponents();
   }));
 
   beforeEach(() => {
     fixture = TestBed.createComponent(TestTableComponent);
     testComponent = fixture.componentInstance;
+    jobs = testJobs();
+    testComponent.jobs.next(jobs);
   });
 
   it('should display a row for each job', async(() => {
     fixture.detectChanges();
     let de: DebugElement = fixture.debugElement;
-    expect(de.queryAll(By.css('.mat-row')).length).toEqual(testJobs.length);
+    expect(de.queryAll(By.css('.mat-row')).length).toEqual(jobs.length);
   }));
 
   it('should display general job data in row', async(() => {
-    testComponent.jobs.next({
-      results: [testJob1],
-      exhaustive: true
-    });
+    testComponent.jobs.next([jobs[0]]);
     fixture.detectChanges();
     let de: DebugElement = fixture.debugElement;
     expect(de.query(By.css('.job-details-button')).nativeElement.textContent)
-      .toEqual(testJob1.name);
+      .toEqual(jobs[0].name);
     expect(de.query(By.css('#submitted-column')).nativeElement.textContent)
       .toContain('8:00 PM');
     expect(de.queryAll(By.css('.additional-column')).length).toEqual(0);
   }));
 
   it('should display dsub-specific job data in row', async(() => {
     environment.additionalColumns = ['user-id', 'status-detail'];
-    testComponent.jobs.next({
-      results: [testJob1],
-      exhaustive: true
-    });
+    testComponent.jobs.next([jobs[0]]);
     fixture.detectChanges();
     let de: DebugElement = fixture.debugElement;
     expect(de.query(By.css('.job-details-button')).nativeElement.textContent)
-      .toEqual(testJob1.name);
+      .toEqual(jobs[0].name);
     let dsubColumns = de.queryAll(By.css('.additional-column'));
     expect(dsubColumns.length).toEqual(2);
     expect(dsubColumns[0].nativeElement.textContent)
-      .toEqual(testJob1.labels['user-id']);
+      .toEqual(jobs[0].labels['user-id']);
     expect(dsubColumns[1].nativeElement.textContent)
-      .toEqual(testJob1.labels['status-detail']);
+      .toEqual(jobs[0].labels['status-detail']);
   }));
 
   it('displays the abort button for selected active jobs', async(() => {
```

Comment:
Regardless of whether you decide to split out the test I mentioned in job-list-spec above, here I think it does make sense to have these be 4 separate unit tests.

Reply:
Done